### PR TITLE
pkgs.nixos: Add example to inline documentation

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Usage of the NixOS function may not be obvious; in particular getting things out that are not already in `system.build`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -22097,38 +22097,53 @@ with pkgs;
...
+         let
+           myOS = pkgs.nixos ({ lib, pkgs, config, ... }: {
+
+             config.services.nginx = {
+               enable = true;
+               # ...
+             };
+
+             # This exports a runner for nginx
+             config.system.build.run-nginx = config.systemd.services.nginx.runner;
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: This is the first time I see runner. This is well-kept secret.

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: I guess this is very useful when building systemd-less container.

Comment (arianvp) [reply pkgs/top-level/all-packages.nix]: The option isn't documented in  `man configuration.nix` for some reason

Comment (arianvp) [reply pkgs/top-level/all-packages.nix]: It's in `nixos/modules/testing/service-runner.nix` by the way for the curious cat!

Comment (arianvp) [reply pkgs/top-level/all-packages.nix]: However, it only seems to implement a very small subset of features systemd has... Clearly meant for testing. Only a small subset of. `man systemd.exec` and `man systemd.service` options is actually used.

**Review by srhb — APPROVED** [review]

**Pull request merged**